[Solutions] add e2e test for solution ui intergration

ChengYanJin opened the pull request:
**Component**: ui,solution, e2e test

**Context**: 
We need to implement the cypress test for the solution UI. To test the following scenarios:

- Test creation of Environment (simple form)
- Test addition of a Solution to an Environment (validate URL redirect)
- Test deployment of a Solution after the preparation is complete
- Test deletion of Environment

We should be able to run the different scenarios separately.

**Summary**:
Implement a Cypress test suite for Environment creation and preparation from Platform UI

**Acceptance criteria**: 
need to merge #1850 first so that the CI is ready.

- comment to execute the cypress test locally
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/solutions/CreateEnvironment.feature --no-exit`
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/solutions/AddSolutionToEnvironment.feature --no-exit`
`./node_modules/cypress/bin/cypress run -b chrome -s cypress/integration/solutions/DeleteEnvironment.feature --no-exit`

- Since we want to run the solution end to end test in CI in a separate VM to avoid flaky, we should be able to execute the solution frontend e2e separately by the following comment.
`npm run test:e2esolutions`

- In the CI, we should execute the solution e2e test by the following comment (Should use terraform to plug in the e2e test)
`npm run test:e2esolutions -- -e target_url="https://${CONTROL_PLANE_IP}:8443"`


Best Practice: Tests should always be able to be run independently from one another and still pass.
Anti-Pattern: Coupling multiple tests together.

---

<!-- Declare one or more issues to close once this PR gets merged -->

Closes: #1851 

<!-- If you want to refer to an issue while not closing it, use:

See: #ISSUE_NUMBER

-->

ChengYanJin committed:
ui/test: Add e2e test for solution ui intergration

Refs: #1851

bert-e commented:
# Hello chengyanjin,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

ChengYanJin committed:
ui/package: Add solution e2e test script

In the CI, we want to run the solution e2e test in a separate VM to avoid the flaky.
So we need to have a dedicated script to execute only the solution e2e test.

Refs: #1851

alexandre-allard commented on ui/cypress/integration/Solutions.feature:
```diff
@@ -0,0 +1,10 @@
+   Scenario: Environment List
+     Given I log in
+     When I go to the solutions list by clicking the solution icon in the sidebar
+     And I go to the Create Envrionments page by clicking the Create a New Environment button
```
```suggestion
     And I go to the Create Environments page by clicking the Create a New Environment button
```

ChengYanJin replied:
Sorry for the spelling mistakes.
And thank you for the correction!

alexandre-allard commented on ui/cypress/integration/common/solutions.js:
```diff
@@ -0,0 +1,121 @@
+);
+
+And(
+  'I go to the Create Envrionments page by clicking the Create a New Environment button',
```
```suggestion
  'I go to the Create Environments page by clicking the Create a New Environment button',
```

alexandre-allard commented on ui/cypress/integration/common/solutions.js:
```diff
@@ -0,0 +1,121 @@
+);
+
+Then(
+  'I fill out the Create Envrionments form and check if the environment I created is displayed on the Environments list',
```
```suggestion
  'I fill out the Create Environments form and check if the environment I created is displayed on the Environments list',
```

ChengYanJin committed:
ui/e2etest: Split the big scenario into three

- We should be able to execute each of the scenario separately
- Group the feature relate to solution inside the solutions fold, others in common fold

Refs: #1851

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/1851-add-e2e-test-of-solution-ui-interation` and the
destination branch `development/2.6`.

Please resolve the conflict on **the feature branch** (`improvement/1851-add-e2e-test-of-solution-ui-interation`).

```bash
 $ git fetch
 $ git checkout origin/improvement/1851-add-e2e-test-of-solution-ui-interation
 $ git merge origin/development/2.6
 $ # <intense conflict resolution>
 $ git commit
 $ git push origin HEAD:improvement/1851-add-e2e-test-of-solution-ui-interation
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @tcarmet

* @jbertran

* @carlito-scality

* @slaperche-scality

* @nicolast

* @teddyandrieux

* @monpote

* @sayf-eddine-scality

* @nootal

* @alexandre-allard-scality

* @gdemonet

* @lucieleonard

* @thomasdanan

* @cmonfort

MonPote reviewed (approved):
LGTM

ChengYanJin commented:
/approve

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`
* `development/2.4`
* `development/2.5`



Please check the status of the associated issue None.

Goodbye chengyanjin.
